src/test/store/receive-profile.test.js

```diff
@@ -436,6 +442,57 @@ describe('actions/receive-profile', function() {
         '  - hide [thread MediaDecoderStateMachine idle]',
       ]);
     });
+    describe('getRangeFilteredScreenshotsById', function() {
+      it('can extract some network markers and match the snapshot', function() {
+        const profile = getScreenshotTrackProfile();
+        const { getState } = storeWithProfile(profile);
+        const screenshotMarkersById = selectedThreadSelectors.getRangeFilteredScreenshotsById(
+          getState()
+        );
+        const keys = [...screenshotMarkersById.keys()];
+        expect(keys.length).toEqual(2);
+
+        const [screenshots] = [...screenshotMarkersById.values()];
+        if (!screenshots) {
+          throw new Error('No screenshots found.');
+        }
+        // The profile contains the same markers twice, but with different window ids.
+        // `screenshots` contains the screenshots for only one window id.
+        // That's why we need to halve the markers length when comparing the 2 lengths.
+        expect(screenshots.length).toEqual(
+          profile.threads[0].markers.length / 2
+        );
+        for (const screenshot of screenshots) {
+          expect(screenshot.name).toEqual('CompositorScreenshot');
+        }
+      });
+
+      it('only deliveres screenshots within a range selection', function() {
+        const profile = getScreenshotTrackProfile();
+        const [{ markers }] = profile.threads;
+        const { dispatch, getState } = storeWithProfile(profile);
+
+        // Double check that there are 10 markers in the test data, and commit a
+        // subsection of that range.
+        expect(markers.length).toBe(20);
+        const startIndex = 3;
+        const endIndex = 8;
+        const startTime = ensureExists(markers.startTime[startIndex]);
+        const endTime = ensureExists(markers.startTime[endIndex]) - 0.1;
+        dispatch(ProfileView.commitRange(startTime, endTime));
+
+        // Get out the markers.
+        const screenshotMarkersById = selectedThreadSelectors.getRangeFilteredScreenshotsById(
+          getState()
+        );
+        const [key] = [...screenshotMarkersById.keys()];
+        const screenshots = screenshotMarkersById.get(key);
+        if (!screenshots) {
+          throw new Error('No screenshots found.');
+        }
+        expect(screenshots.length).toEqual(endIndex - startIndex + 1);
+      });
+    });
```

Comment:
Hi @canova, I'm still not sure how to write tests. I copied this from profile-view.test.js. This works even removing the changes I made in the merge-compare.js file. I guess this is not the way it should be done.